### PR TITLE
[opt](Nereids) support column permission control

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

add a rewriter in "Final rewrite and check" to check col auth

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AuthChecker.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        if (catalog == null) {
+            return;
+        }
+        if (cols.size() == 0) {
+            return;
+        }
+        try {
+            Env.getCurrentEnv().getAccessManager()
+                    .checkColumnsPriv(ConnectContext.get().getCurrentUserIdentity(), catalog.getName(),
+                            database.getFullName(), table.getName(), cols, PrivPredicate.SELECT);
```
[the remark this answers is not in the document]
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AuthChecker.java]: not this method

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.45 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162412279 Bytes

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.77 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162409436 Bytes

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run feut

**Comment by zddr** [comment]
run feut

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run buildall

**Review by morningman — COMMENTED** [review]
Need more complex case

**Comment by zddr** [comment]
由于视图问题，暂时不加列权限

**Pull request closed without merging**